### DISCUSSION TITLE
Filter: Add help text to FilterSet

### DISCUSSION THREAD

**Discussion by snowman2** [encode/django-rest-framework#7967] [Ideas & Suggestions]
What are your thoughts on adding optional `help_text` to the `FilterSet`. 

Here is an example of how we use it:

```pyhton
class AttributeFilter(FilterSet):
    class Meta(object):
        model = Attribute
        fields = {
            "id": ["exact", "in"],
            "name": ["exact", "iexact", "icontains", "contains"],
        }
        help_text = {
            "id": {
                "exact": "id/primary key for an Attribute entry.",
                "in": "Comma separated list of ids/primary keys for a set of Attribute entries.",
            },
            "name": {
                "exact": "Attribute name field.",
                "iexact": "Case insensitive search for attribute name field.",
                "icontains": "Case insensitive `like` search for Attribute names.",
                "contains": "Case sensitive `like` search for Attribute names.",
            },
        }
```

This is useful when auto-generating API documentation.

cc: @alfredoahds

Note: I should add that we would be willing to submit a PR if you would like to see this added.

**Comment by jerinpetergeorge** [comment]
If I'm not wrong, the `FilterSet` comes from the [django-filter](https://github.com/carltongibson/django-filter), a separate package.

btw, I usually don't care about the help-texts in APIs. But, it can be handy when you render them in your native Django or DRF-Browsable API.

**Reply by snowman2** [reply]
Oof, you're right about it being in django-filter. Apologies for the noise.
